helix-term/src/grammars.rs

```diff
@@ -54,97 +61,77 @@ pub fn fetch_grammar(grammar: GrammarConfiguration) {
 
         // create the grammar dir contains a git directory
         if !grammar_dir.join(".git").is_dir() {
-            Command::new("git")
-                .args(["init"])
-                .current_dir(grammar_dir.clone())
-                .output()
-                .expect("Could not execute 'git'");
+            git(&grammar_dir, ["init"])?;
         }
 
         // ensure the remote matches the configured remote
         if get_remote_url(&grammar_dir).map_or(true, |s| s.trim_end() != remote) {
-            set_remote(&grammar_dir, &remote);
+            set_remote(&grammar_dir, &remote)?;
         }
 
         // ensure the revision matches the configured revision
-        if get_revision(&grammar_dir).map_or(true, |s| s != revision) {
+        if get_revision(&grammar_dir).map_or(true, |s| s.trim_end() != revision) {
```

Comment:
It looks like `get_remote_url` and `get_revision` both swallow the error. This is again a little hard to tell what all the error conditions are, but if this does fail, it will just go to the else and say that the grammar is already up to date.

Reply:
Oh I see, yeah, I got the logic reversed. But... If we failed to get the current revision, will fetching and checking out work? Would it be better to return the error early and only proceed if we successfully passed the head, but it was not equal?

Reply:
The rev-parse errors out if the repository is empty with no commits which is the case right after the `git init`, so I want that case to be covered by fetching from the remote and checking it out.

There might be some cases where it's failing for some reason other than being empty but I would expect those failures to pop up in fetching/setting the remote as well

Reply:
Yeah, probably so. :+1: